### DISCUSSION TITLE
Do we have event handling in Liquidsoap?

### DISCUSSION THREAD

**Discussion by Moonbase59** [savonet/liquidsoap#3961] [Q&A] [answered]
For autocue, I’m thinking of something that could "fire an event" when cue_file is finished with a track, and pass on some metadata.

So a third-party script could use something like `radio.on_cue_file` maybe, to pick up changed metadata and update their database. Like AzuraCast, for instance.

Is that possible with what we have (i.e. LS 2.2.5)?

**Comment by toots** [comment, marked as answer]
We have event-based APIs but there is no generic event API available at the moment, no.

**Comment by Moonbase59** [comment]
Hm. So if I wanted AzuraCast to be able to pick up calculated Autocue values for its database, I’d have to put in something App-specific into otherwise "pure" Liquidsoap code. Not so elegant…

I had somehow hoped for user-generated events… ;-)

**Reply by toots** [reply]
It's not hard to add but for large features like that it's good to think a little more about the general picture:
* What would be the use-cases it would solve that can definitely not be solved conveniently at the moment
* What can we do with the type system to make it robust, typically exposing the set of events any given value can send or receive
* Will that impact performances, memory and CPU usage
